cmd/clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -92,29 +92,22 @@ func (d *ClusterDeployer) Create(cluster *clusterv1.Cluster, machines []*cluster
 	}
 	defer closeClient(bootstrapClient, "bootstrap")
 
-	if cluster.Namespace == "" {
-		cluster.Namespace = bootstrapClient.GetContextNamespace()
-	}
-
-	err = bootstrapClient.EnsureNamespace(cluster.Namespace)
-	if err != nil {
-		return fmt.Errorf("unable to ensure namespace %q in bootstrap cluster: %v", cluster.Namespace, err)
-	}
-
 	glog.Info("Applying Cluster API stack to bootstrap cluster")
 	if err := phases.ApplyClusterAPIComponents(bootstrapClient, d.providerComponents); err != nil {
```

Comment:
ApplyClusterAPIComponents doesn't include the `client.EnsureNamespace(namespace)` call which used to proceed it. Will it now fail if the namespace doesn't exist? Does it also need to include that call in it's phase?

Reply:
sgtm to remove it for now, and we can re-add if we need to.